src/mpid/ch4/shm/ipc/src/ipc_pre.h

```diff
@@ -6,15 +6,11 @@
 #ifndef IPC_PRE_H_INCLUDED
 #define IPC_PRE_H_INCLUDED
 
-#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
 #include "../xpmem/xpmem_pre.h"
-#endif
 
 typedef enum {
     MPIDI_SHM_IPC_TYPE__NONE,   /* avoid empty enum */
-#ifdef MPIDI_CH4_SHM_ENABLE_XPMEM
     MPIDI_SHM_IPC_TYPE__XPMEM,
-#endif
 } MPIDI_SHM_IPC_type_t;
```

Comment:
It looks inconsistent that we disable the XPMEM mem handle routine in IPC but always  enable the type definition. Since Pavan does not want to see so many guards check in IPC, how about we define a `xpmem_stub.h` and provide empty definitions for all exposed XPMEM function/type?

We will need to update xpmem/Makefile.am too to ensure the appropriate file(s) are compiled.

Reply:
We always include `xpmem_pre.h` so it does we can avoid guarding the data structure definitions. So I just guarded `xpmem.h` in `xpmem_pre.h`. For init/finalize we can do that sure. Other functions are still guarded, but I guess it is fine for those as this only happens in one place (`ipc_mem.h`)